Add loading state when viewing commit changes in Insomnia designer

**Is your feature request related to a problem? Please describe.**
As a user I want to be informed that my changes are still loading so that I know whats going on (and don't close the window before the changes load)

**Describe the solution you'd like**
Add any old loading spinner while the changes are loading up or some sick elevator music 🎵 :trollface: 

**Describe alternatives you've considered**
Waiting but who does that these days?

**Additional context**
![Jul-28-2020 13-01-00](https://user-images.githubusercontent.com/182836/88607087-6ede6680-d0d2-11ea-9c20-c2050f07cb2e.gif)

Sorry for the delay but I'm still facing issues. This one in particular: 

```bash
lerna ERR! npm install exited 1 in 'insomnia-inso'
lerna ERR! npm install stderr:

┌────────────────────────────────────────────────────────────┐
│                  npm update check failed                   │
│            Try running with sudo or get access             │
│            to the local update config store via            │
│ sudo chown -R $USER:$(id -gn $USER) C:\Users\abhih\.config │
└────────────────────────────────────────────────────────────┘
npm WARN tarball tarball data for flow-bin@0.126.1 (the SHA checksum here) seems to be corrupted. Trying one more time.
npm WARN tarball tarball data for flow-bin@0.126.1 (the SHA checksum here) seems to be corrupted. Trying one more time.
```
Then it gives a bunch of `npm ENOENT` errors and displays this:
```bash
npm ERR! code EINTEGRITY
npm ERR! Verification failed while extracting flow-bin@0.126.1
npm ERR! Verification failed while extracting flow-bin@0.126.1
npm ERR! [an SHA checksum] integrity checksum failed when using sha512: wanted [another SHA checksum]
```
and then again a bunch of `ENOENT` errors and then finally crashes with this.
```bash
lerna ERR! npm install exited 1 in 'insomnia-inso'
lerna WARN complete Waiting for 7 child processes to exit. CTRL-C to exit immediately.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! insomnia@1.0.0 bootstrap: `npm install && lerna bootstrap && lerna run --parallel --stream bootstrap`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the insomnia@1.0.0 bootstrap script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

I've installed Windows build tools. I've tried verifying and cleaning the `npm` cache.  
I think this is a proxy issue on my side as mentioned [here](https://github.com/npm/npm/issues/20925). But running `npm config get proxy` returned `null`. I'm aware that this is an `npm` issue but any pointers will be helpful.

Sorry again for the delayed updates. I've already spent a good deal of time understanding these `npm` errors and yet I face a new one every time. I'm trying to solve the "real" issue here but cannot guarantee any quick updates as I'm facing issues setting up the development environment locally.

P.S. If this is an urgent update and I'm hindering the app's development flow, then anyone else can take it up. =)

--- a/packages/insomnia-app/app/ui/components/modals/git-staging-modal.js
+++ b/packages/insomnia-app/app/ui/components/modals/git-staging-modal.js
@@ -417,7 +417,7 @@ class GitStagingModal extends React.PureComponent<Props, State> {
             <button className="btn" onClick={this._hideModal}>
               Close
             </button>
-            <button className="btn" onClick={this._handleCommit} disabled={!hasChanges}>
+            <button className="btn" onClick={this._handleCommit} disabled={loading || !hasChanges}>
               Commit
             </button>
           </div>

--- a/packages/insomnia-app/app/ui/components/modals/git-staging-modal.js
+++ b/packages/insomnia-app/app/ui/components/modals/git-staging-modal.js
@@ -32,6 +32,7 @@ type Item = {|
 |};
 
 type State = {|
+  loading: boolean,
   branch: string,
   message: string,
   items: {
@@ -40,6 +41,7 @@ type State = {|
 |};
 
 const INITIAL_STATE: State = {
+  loading: false,
   branch: '',
   message: '',
   items: {},
@@ -173,6 +175,12 @@ class GitStagingModal extends React.PureComponent<Props, State> {
   async _refresh(callback?: () => void) {
     const { vcs, workspace } = this.props;
 
+    this.setState({ loading: true });
+
+    // Get and set branch name
+    const branch = await vcs.getBranch();
+    this.setState({ branch });
+
     // Cache status names
     const docs = await withDescendants(workspace);
     const allPaths = await this.getAllPaths();
@@ -239,8 +247,7 @@ class GitStagingModal extends React.PureComponent<Props, State> {
       };
     }
 
-    const branch = await vcs.getBranch();
-    this.setState({ items, branch }, callback);
+    this.setState({ items, loading: false }, callback);
   }
 
   renderOperation(item: Item) {
@@ -355,64 +362,68 @@ class GitStagingModal extends React.PureComponent<Props, State> {
     );
   }
 
-  render() {
-    const { items, message, branch } = this.state;
+  _renderEmpty() {
+    const { loading } = this.state;
 
-    const itemsList = Object.keys(items).map(k => items[k]);
-    let body = null;
-    if (itemsList.length === 0) {
-      body = (
-        <>
-          <ModalHeader>Commit Changes</ModalHeader>
-          <ModalBody className="wide pad">No changes to commit.</ModalBody>
-          <ModalFooter>
-            <div className="margin-left italic txt-sm tall">
-              <i className="fa fa-code-fork" /> {branch}
-            </div>
-            <div>
-              <button className="btn" onClick={this._hideModal}>
-                Close
-              </button>
-            </div>
-          </ModalFooter>
-        </>
-      );
-    } else {
-      const newItems = itemsList.filter(i => i.status.includes('added'));
-      const existingItems = itemsList.filter(i => !i.status.includes('added'));
-
-      body = (
-        <>
-          <ModalHeader>Commit Changes</ModalHeader>
-          <ModalBody className="wide pad">
-            <div className="form-control form-control--outlined">
-              <textarea
-                ref={this._setTextareaRef}
-                rows="3"
-                required
-                placeholder="A descriptive message to describe changes made"
-                defaultValue={message}
-                onChange={this._handleMessageChange}
-              />
-            </div>
-            {this.renderTable('Modified Objects', existingItems, 'Rollback all')}
-            {this.renderTable('Unversioned Objects', newItems, 'Delete all')}
-          </ModalBody>
-          <ModalFooter>
-            <div className="margin-left italic txt-sm tall">
-              <i className="fa fa-code-fork" /> {branch}
-            </div>
-            <div>
-              <button className="btn" onClick={this._handleCommit}>
-                Commit
-              </button>
-            </div>
-          </ModalFooter>
-        </>
-      );
+    if (loading) {
+      return <>Loading...</>;
     }
 
-    return <Modal ref={this._setModalRef}>{body}</Modal>;
+    return <>No changes to commit.</>;
+  }
+
+  _renderItems(items: Array<Item>) {
+    const { message } = this.state;
+
+    const newItems = items.filter(i => i.status.includes('added'));
+    const existingItems = items.filter(i => !i.status.includes('added'));
+
+    return (
+      <>
+        <div className="form-control form-control--outlined">
+          <textarea
+            ref={this._setTextareaRef}
+            rows="3"
+            required
+            placeholder="A descriptive message to describe changes made"
+            defaultValue={message}
+            onChange={this._handleMessageChange}
+          />
+        </div>
+        {this.renderTable('Modified Objects', existingItems, 'Rollback all')}
+        {this.renderTable('Unversioned Objects', newItems, 'Delete all')}
+      </>
+    );
+  }
+
+  render() {
+    const { items, branch, loading } = this.state;
+
+    const itemsList = Object.keys(items).map(k => items[k]);
+    const hasChanges = !!itemsList.length;
+
+    return (
+      <Modal ref={this._setModalRef}>
+        <ModalHeader>Commit Changes</ModalHeader>
+        <ModalBody className="wide pad">
+          {hasChanges ? this._renderItems(itemsList) : this._renderEmpty()}
+        </ModalBody>
+        <ModalFooter>
+          <div className="margin-left italic txt-sm tall">
+            <i className="fa fa-code-fork" /> {branch}{' '}
+            {loading && <i className="fa fa-refresh fa-spin" />}
+          </div>
+          <div>
+            <button className="btn" onClick={this._hideModal}>
+              Close
+            </button>
+            <button className="btn" onClick={this._handleCommit} disabled={!hasChanges}>
+              Commit
+            </button>
+          </div>
+        </ModalFooter>
+      </Modal>
+    );
   }
 }
 

